### PR TITLE
BUG: Fix nansum overflow on Windows with bottleneck

### LINKED ISSUE #15453: BUG: int64 overflow/wrap around with sum()
#### Code Sample, a copy-pastable example if possible

```python
In [1]: import pandas as pd

In [2]: s = pd.Series([2**31])

In [3]: print(s.dtype, s.sum())
(dtype('int64'), -2147483648)

In [4]: pd.Series([2**31 - 1, 1]).sum()
Out[4]: -2147483648

In [5]: pd.Series([2**31 - 1, 1]).astype('int32').sum()
Out[5]: 2147483648
```

#### Problem description

negative values in [3] and [4]

#### Expected Output

see [5]

#### Output of ``pd.show_versions()``

<details>

```python
In [6]: pd.show_versions()

INSTALLED VERSIONS
------------------
commit: None
python: 2.7.13.final.0
python-bits: 64
OS: Windows
OS-release: 7
machine: AMD64
processor: Intel64 Family 6 Model 42 Stepping 7, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.19.2
nose: 1.3.7
pip: 9.0.1
setuptools: 34.2.0
Cython: None
numpy: 1.12.0
scipy: 0.19.0rc1
statsmodels: 0.8.0
xarray: None
IPython: 5.2.2
sphinx: 1.5.2
patsy: None
dateutil: 2.6.0
pytz: 2016.10
blosc: None
bottleneck: 1.2.0
tables: None
numexpr: 2.6.2
matplotlib: 1.5.3
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: 4.5.3
html5lib: 0.999999999
httplib2: 0.10.3
apiclient: 1.6.2
sqlalchemy: 1.1.5
pymysql: None
psycopg2: 2.6.2 (dt dec pq3 ext lo64)
jinja2: 2.9.5
boto: None
pandas_datareader: None
```
</details>

### PR DESCRIPTION
**Pull request by JaviSorribes**
Bottleneck used in nansum produces overflow error on Windows. Temporary fix that could be undone once the error in bottleneck is fixed and released, to avoid platform specificity.

 - [x] closes #15453 
 - [x] tests added / passed
 - [x] passes ``git diff upstream/master | flake8 --diff``
 - [x] whatsnew entry

### CODE REVIEW

File: pandas/core/nanops.py [hunk 1]
```diff
@@ -130,6 +131,8 @@ def _bn_ok_dtype(dt, name):
         if name == 'nansum':
             if dt.itemsize < 8:
                 return False
+            elif os.name == 'nt':  # bottleneck has overflow issue in Windows
```
Comment (jreback) [inline pandas/core/nanops.py]: use ``pandas.compat.is_platform_windows()``

File: pandas/tests/test_nanops.py [hunk 2]
```diff
@@ -327,6 +327,12 @@ def test_nansum(self):
         self.check_funs(nanops.nansum, np.sum, allow_str=False,
                         allow_date=False, allow_tdelta=True, check_dtype=False)
 
+    def test_nansum_overflow(self):
+        s = Series([2**31])
```
Comment (jreback) [inline pandas/tests/test_nanops.py]: needs testing both with and w/o bottleneck (to yield the same results), see other tests for how to do this.

**Review by jreback — CHANGES_REQUESTED** [review]
The biggest issue here is this semantically changes things. IOW, now since we are *always* using pandas nansum (rather than bottleneck nansum if installed), we will *always* yield ``Series([np.nan]).sum()`` -> ``np.nan`` (rather than 0 if bottleneck is installed). So I think we actually have to change this for *all* platforms or then this becomes even more confusing.

**Comment by jreback** [comment]
@jorisvandenbossche @shoyer 

w.r.t. #9422 I think we need to simply change this (in effect what this PR does), but for all platforms (and not just windows).

**Comment by cpcloud** [comment]
@JaviSorribes Looking at @jreback's comments, it looks like we should just disable `nansum` bottleneck calls altogether.

**Comment by JaviSorribes** [comment]
So I understand @jreback and @cpcloud that we should not use bottleneck at all for nansum? Checkout the image attached for what I think this would look like.

Also, what about all the other nan operations? Should we consider possible overflow in any of them?

image

**Comment by jreback** [comment]
xref #15542

**Comment by jorisvandenbossche** [comment]
We indeed just need to take a decision on #9422, as we have left this lingering too long. 
But, I am not necessarily convinced that we should change everything to the native pandas way (returning NaN), instead of changing pandas to return 0 to align with numpy/bottleneck (and also R, matlab, ..). 
I don't think there was really agreement on this in #9422 (several have argued for both positions), but I also don't know how to best proceed on taking a decision.

**Comment by jreback** [comment]
The fixes are related to fixing #9422 are a bit bigger in scope than this. That said If you want to rebase / add some additional tests could take this as a precurser.

**Comment by jreback** [comment]
closing this. as #9422 needs resolution.

**Pull request closed without merging**